cpp/modmesh/buffer/pymod/wrap_SimpleArray.cpp

```diff
@@ -75,15 +75,19 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
                             throw std::runtime_error("dtype mismatch");
                         }
                         modmesh::detail::shape_type shape;
+                        modmesh::detail::shape_type stride;
+                        bool is_c_contiguous = (arr_in.flags() & py::array::c_style) == py::array::c_style;
+                        constexpr size_t itemsize = wrapped_type::itemsize();
                         for (ssize_t i = 0; i < arr_in.ndim(); ++i)
                         {
                             shape.push_back(arr_in.shape(i));
+                            stride.push_back(arr_in.strides(i) / itemsize);
                         }
                         std::shared_ptr<ConcreteBuffer> const buffer = ConcreteBuffer::construct(
                             arr_in.nbytes(),
                             arr_in.mutable_data(),
                             std::make_unique<ConcreteBufferNdarrayRemover>(arr_in));
-                        return wrapped_type(shape, buffer);
+                        return is_c_contiguous ? wrapped_type(shape, buffer) : wrapped_type(shape, stride, buffer);
```

Comment:
If the array is c contiguous, the wrapper use the original constructor.

Reply:
f_contiguous also needs to be handled.